packages/frontend-2/components/viewer/views/Menu.vue

```diff
@@ -1,10 +1,8 @@
 <template>
-  <Menu as="div" class="relative z-30">
-    <MenuButton :id="menuButtonId" v-slot="{ open }" as="template">
-      <ViewerControlsButtonToggle flat secondary :active="open">
-        <IconViews class="w-5 h-5" />
-      </ViewerControlsButtonToggle>
-    </MenuButton>
+  <div ref="menuWrapper" class="relative z-30">
```

Comment:
Instead of removing Menu, you could've just used its API to manually manage its open/closed state. The Menu still has accessability foundations baked in, which I'm not sure if you've re-implemented. See MenuItems static prop

Reply:
I did try using the static prop with MenuItems, but it still required wrapping them in a v-show, which caused the open state to toggle when a MenuItem was clicked. They do provide a close() slot prop, but there's no way to keep the menu open on selection.

I also experimented with managing a second state to control the visibility of MenuItems using v-show, but it ended up being a bit of a mess. The accessibility features of Headless UI are tightly integrated with their internal open state. Trying to override this with our own state just didn't work well.

It felt like I was fighting against the library rather than working with it.

Maybe there is something I'm missing that is an easy solution?

Reply:
Nah, if it was a big PITA then skip it. Do ensure its still keyboard-navigatable tho